sound/soc/sof/topology.c

```diff
@@ -2886,60 +2889,73 @@ static int sof_link_ssp_load(struct snd_soc_component *scomp, int index,
 			     struct snd_soc_dai_link *link,
 			     struct snd_soc_tplg_link_config *cfg,
 			     struct snd_soc_tplg_hw_config *hw_config,
-			     struct sof_ipc_dai_config *config)
+			     struct sof_ipc_dai_config *config,
+			     int num_conf, int def_conf)
 {
 	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
 	struct snd_soc_tplg_private *private = &cfg->priv;
+	int common_data = num_conf ? num_conf : 1;
 	u32 size = sizeof(*config);
 	int ret;
+	int i;
 
-	/* handle master/slave and inverted clocks */
-	sof_dai_set_format(hw_config, config);
-
-	/* init IPC */
-	memset(&config->ssp, 0, sizeof(struct sof_ipc_dai_ssp_params));
-	config->hdr.size = size;
+	/*
+	 * Parse common data, we should have 1 common data per hw_config.
+	 * How ever we might not have hw_config at all so parse 1 common data
```

Comment:
spelling: How ever -> However

Reply:
@juimonen I feel like this logic might be flawed. If we have one set of common tokens for all hw_configs, you still need to parse them n times for n hw_configs, so all the configs will be updated right?

Reply:
so either we have 1 common set for all, or every hw_config has its own. I don't know any means to discover which might be the case so it needs to be decided in advance. So in case of similar common data and multiple hw_configs, the common data needs to be replicated -> or we agree that there's always the 1 and same common data. So I was not sure is the common data variating with the hw_config. I guess it could?